File: src/api/types/memo.ts
// 请求参数类型
export interface UploadMemoReq {
    title: string;
    content: string;
    origin_at: number;
    cover_file_id: number;
    profile_content: string;
}

export interface GetMemoDetailReq {
    memo_id: number;
}

export interface GetMemoListReq {
    keyword?: string;
    page: number;
    limit: number;
}

// 响应类型
export interface UploadMemoResp {
    code: number;
    message: string;
    memo_id: number;
}

export interface GetMemoDetailResp {
    code: number;
    message: string;
    memo_detail: MemoDetail;
}

export interface GetMemoListResp {
    code: number;
    message: string;
    memo_profiles: MemoProfile[];
}

// 数据类型
export interface MemoDetail {
    id: number;
    title: string;
    content: string;
    origin_at: number;
    cover_url: string;
    profile_content: string;
}

export interface MemoProfile {
    id: number;
    title: string;
    cover_url: string;
    profile_content: string;
    created_at: number;
    updated_at: number;
    origin_at: number;
}

File: server/biz/prm/memo.go
package prm

import (
	"onij/infra/mysql"
	api "onij/model/api"
	"onij/util"
	"onij/util/boost/collection/collext"
)

type GetMemoDetailParam struct {
	Id int64
}

func NewGetMemoDetailParam(req *api.GetMemoDetailReq) *GetMemoDetailParam {
	return &GetMemoDetailParam{
		Id: req.MemoId,
	}
}

type GetMemoDetailResult struct {
	Memo *mysql.Memo
	CoverFile *mysql.File
}

func (r *GetMemoDetailResult) Resp() *api.GetMemoDetailResp {
	return &api.GetMemoDetailResp{
		Code: util.BaseCodeOK,
		Message: util.BaseMsgOK,
		MemoDetail: &api.MemoDetail{
			Id: r.Memo.Id,
			Title: r.Memo.Title,
			Content: r.Memo.Content,
			OriginAt: r.Memo.OriginAt,
			CoverUrl: util.DownloadFile(r.CoverFile.StoreKey),
			ProfileContent: r.Memo.ProfileContent,
		},
	}
}

type GetMemoListParam struct {
	Keyword string
	util.Page
}

func NewGetMemoListParam(req *api.GetMemoListReq) *GetMemoListParam {
	return &GetMemoListParam{
		Keyword: req.Keyword,
		Page: util.Page{
			Page: req.Page,
			Limit: req.Limit,
		},
	}
}

type GetMemoListResult struct {
	Memos []*mysql.Memo
	CoverFileMap map[int64]*mysql.File
}

func (r *GetMemoListResult) Resp() *api.GetMemoListResp {
	return &api.GetMemoListResp{
		Code: util.BaseCodeOK,
		Message: util.BaseMsgOK,
		Memos: collext.Pick(r.Memos, func(m *mysql.Memo) *api.MemoProfile {
			return &api.MemoProfile{
				Id: m.Id,
				Title: m.Title,
				CoverUrl: util.DownloadFile(r.CoverFileMap[m.CoverFileId].StoreKey),
				ProfileContent: m.ProfileContent,
				CreatedAt: m.CreatedAt.Unix(),
				UpdatedAt: m.UpdatedAt.Unix(),
				OriginAt: m.OriginAt,
			}
		}),
	}
}

type UploadMemoParam struct {
	Title string
	Content string
	OriginAt int64
	CoverFileId int64
	ProfileContent string
}

func NewUploadMemoParam(req *api.UploadMemoReq) *UploadMemoParam {	
	return &UploadMemoParam{
		Title: req.Title,
		Content: req.Content,
		OriginAt: req.OriginAt,
		CoverFileId: req.CoverFileId,
		ProfileContent: req.ProfileContent,
	}
}

type UploadMemoResult struct {
	Id int64
}

func (r *UploadMemoResult) Resp() *api.UploadMemoResp {
	return &api.UploadMemoResp{
		Code: util.BaseCodeOK,
		Message: util.BaseMsgOK,
		MemoId: r.Id,
	}
}